src/wp-includes/media.php

```diff
@@ -1137,16 +1137,6 @@ function wp_get_attachment_image( $attachment_id, $size = 'thumbnail', $icon = f
 			}
 		}
 
-		// Adds 'auto' to the sizes attribute if applicable.
```

Comment:
By adding a default filter to `wp_get_attachment_image`, we can consolidate all of the logic for this feature in `wp_img_tag_add_auto_sizes()`, which is much more maintainable.

Reply:
I've reverted this idea in f07f4d1e, though I still think we should consider doing it here.